samples/affinidi-nextjs-nextauthjs/src/pages/credential-revocation.tsx

```diff
@@ -0,0 +1,174 @@
+): Promise<FlowData> => {
+  const response = await fetch(
+    "/api/issuance/change-credential-status?" +
+      new URLSearchParams({ issuanceConfigurationId }),
+    {
+      method: "POST",
+      body: JSON.stringify({ issuanceRecordId, changeReason }),
+    }
+  );
+
+  if (response.status === 200) {
+    const res = await response.json();
+    flowDataRecordsQuery.refetch();
+    return res;
+  } else {
+    throw Error(await response.text());
+  }
+};
+
+const fetchIssuanceConfigurations = (): Promise<SelectOption[]> =>
+  fetch("/api/issuance/configuration-options", { method: "GET" }).then((res) =>
+    res.json()
+  );
+
+export const getServerSideProps = (async () => {
+  return { props: { featureAvailable: personalAccessTokenConfigured() } };
+}) satisfies GetServerSideProps<{ featureAvailable: boolean }>;
+
+export default function CredentialRevocation({
+  featureAvailable,
+}: InferGetServerSidePropsType<typeof getServerSideProps>) {
+  const [nextToken, setNextToken] = useState<string>();
+  const [tokenStack, setTokenStack] = useState<string[]>([]);
+  const { data: session } = useSession();
+
+  const configurationsQuery = useQuery({
+    queryKey: ["issuanceConfigurations"],
+    queryFn: fetchIssuanceConfigurations,
+    enabled: !!featureAvailable,
+  });
+
+  const configurationId = configurationsQuery.data?.[0]?.value as string;
+
+  const flowDataRecordsQuery = useQuery({
+    queryKey: ["flowDataRecords", configurationId, nextToken],
+    queryFn: ({ queryKey }) => {
+      const [, issuanceConfigurationId, exclusiveStartKey] = queryKey;
+      return fetchIssuanceDataRecords(
+        issuanceConfigurationId as string,
+        exclusiveStartKey as string
+      );
+    },
+    enabled: !!featureAvailable && !!configurationId,
+  });
+
+  const handleNextClick = async () => {
+    if (flowDataRecordsQuery.data?.lastEvaluatedKey) {
+      setNextToken(flowDataRecordsQuery.data?.lastEvaluatedKey);
+      setTokenStack((prevStack) => [
+        ...prevStack,
+        flowDataRecordsQuery.data?.lastEvaluatedKey as string,
+      ]);
+    }
+  };
+
+  const handlePreviousClick = async () => {
+    if (tokenStack.length > 1) {
+      const previousToken = tokenStack[tokenStack.length - 2];
+      setTokenStack((prevStack) => prevStack.slice(0, -1)); // Remove the last token
+      setNextToken(previousToken);
+    } else {
+      setTokenStack((prevStack) => prevStack.slice(0, -1));
+      setNextToken(undefined);
+    }
+  };
+  const hasErrors = !featureAvailable || !session || !session.userId;
+  const renderErrors = () => {
+    if (!featureAvailable) {
+      return (
+        <div>
+          Feature not available. Please set your Personal Access Token in your
+          environment secrets.
+        </div>
+      );
+    }
+
+    if (!session || !session.userId) {
+      return (
+        <div>
+          You must be logged in to issue credentials to your Affinidi Vault
+        </div>
+      );
+    }
+  };
```

Comment:
same here, but maybe you want to do that in another PR as this one is already taking long to be merged.